draft-ietf-mls-protocol.md

```diff
@@ -1435,48 +1461,219 @@ all arrive at the following state:
+before performing decryption.
+
+The signature field allows strong authentication of messages:
+
+~~~
+    struct {
+        uint8  group[32];
+        uint32 epoch;
+        uint32 generation;
+        uint32 sender;
+        opaque content<0..2^32-1>;
+    } MLSSignatureContent;
+~~~
 
-Alternatively, the hash of the previous message can also be included as an additional
-field rather than change the encryption key. This allows for a more flexible approach,
-because the receiving party can choose to ignore it (if the value is not known, or if
-transcript security is not required).
+The signature used in the MLSPlaintext is computed over the MLSSignatureContent
+which covers the metadata information about the current state
+of the group (group identifier, epoch, generation and sender's Leaf index)
+to avoid Group participants to impersonate other participants and in order
+to prevent cross-group attacks.
+
+[[ TODO: A preliminary formal security analysis has yet to be performed on
+this authentication scheme.]]
+
+[[ OPEN ISSUE: Should the padding be required for Handshake messages ?
+Can an adversary get more that the position of a participant in the tree
+without padding ? Should the base ciphertext block length be negotiated or
+is is reasonable to allow leaking an interval for the length of the plaintext
+by allowing to send a variable number of ciphertext blocks ? ]]
+
+Application messages SHOULD be padded to provide some resistance
+against traffic analysis techniques over encrypted traffic.
+{{?CLINIC=DOI.10.1007/978-3-319-08506-7_8}}
+{{?HCJ16=DOI.10.1186/s13635-016-0030-7}}
+While MLS might be less susceptible to serve the same payload multiple time across
+a lot of ciphertexts than traditional web servers, it might still provide
+the attacker enough information to mount an attack. If Alice asks Bob:
+"When are we going to the movie ?" the answer "Wednesday" might be leaked
+by the ciphertext length to an adversary expecting Alice to provide Bob
+with a day of the week at some point in their discussion.
+
+Similarly to TLS 1.3, if padding is used, the MLS messages MUST be padded
+before AEAD encryption with zero-valued bytes. Upon AEAD decryption,
+the length field of the plaintext is used to compute the number of bytes
+to be removed from the plaintext to get the correct data.
+As the padding mechanism is used to improve protection against traffic
+analysis, removal of the padding SHOULD be implemented in a "constant-time"
+manner at the MLS layer and above to prevent timing side-channels that
+would provide attackers with information on the size of the plaintext.
+
+### Delayed and Reordered Application messages
+
+Since each Application message contains the Group identifier, the epoch and a
+message counter, a participant can receive messages out of order.
+Participants can decrypt these messages if they are able to retrieve
+or recompute the correct AEAD decryption key from currently stored
+cryptographic material.
```

Comment:
If they are able to retrieve ... participants can decrypt

Reply:
Fixed, Thanks !